WordPress/Classes/ViewRelated/Menus/MenuItemAbstractView.m

```diff
@@ -0,0 +1,379 @@
+                                              widthConstraint,
+                                              heightConstraint
+                                              ]];
+    [self addAccessoryButton:button];
+    
+    return button;
+}
+
+- (UIColor *)contentViewBackgroundColor
+{
+    UIColor *color = nil;
+    if (self.highlighted) {
+        color = [WPStyleGuide mediumBlue];
+    } else  {
+        color = [UIColor whiteColor];
+    }
+    
+    return color;
+}
+
+- (UIColor *)textLabelColor
+{
+    UIColor *color = nil;
+    if (self.highlighted) {
+        color = [UIColor whiteColor];
+    } else  {
+        color = [WPStyleGuide darkGrey];
+    }
+    
+    return color;
+}
+
+- (UIColor *)iconTintColor
+{
+    UIColor *color = nil;
+    if (self.highlighted) {
+        color = [UIColor whiteColor];
+    } else  {
+        color = [WPStyleGuide grey];
+    }
+    
+    return color;
+}
+
+#pragma mark - touches
+
+- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
+{
+    [super touchesBegan:touches withEvent:event];
+    self.highlighted = YES;
+}
+
+- (void)touchesMoved:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
+{
+    [super touchesMoved:touches withEvent:event];
+    self.highlighted = NO;
+}
+
+- (void)touchesEnded:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
+{
+    [super touchesEnded:touches withEvent:event];
+    self.highlighted = NO;
+}
+
+- (void)touchesCancelled:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
+{
+    [super touchesCancelled:touches withEvent:event];
+    self.highlighted = NO;
+}
+
+#pragma mark - overrides
+
+- (void)traitCollectionDidChange:(UITraitCollection *)previousTraitCollection
+{
+    [super traitCollectionDidChange:previousTraitCollection];
+    [self setNeedsDisplay];
+    [self.contentView setNeedsDisplay];
+}
+
+- (void)drawRect:(CGRect)rect
+{    
+    CGContextRef context = UIGraphicsGetCurrentContext();
+    
+    CGRect dashRect = CGRectInset(self.contentView.frame, 8.0, 8.0);
+    
+    CGContextSetStrokeColorWithColor(context, [[WPStyleGuide mediumBlue] CGColor]);
+    CGContextSetLineWidth(context, 1.0);
+    
+    const CGFloat dashLength = 6.0;
+    const CGFloat dashFillPercentage = 60; // fill % of the line with dashes, rest with white space
+    
+    CGFloat(^spacingForLineLength)(CGFloat) = ^ CGFloat (CGFloat lineLength) {
+        // calculate the white spacing needed to fill the full line with dashes
+        const CGFloat dashFill = (lineLength * dashFillPercentage) / 100;
+        //// the white spacing is proportionate to amount of space the dashes will take
+        //// uses (dashFill - dashLength) to ensure there is one extra dash to touch the end of the line
+        return ((lineLength - dashFill) * dashLength) / (dashFill - dashLength);
+    };
+    
+    const CGFloat pointOffset = 0.5;
+    {
+        CGFloat dash[2] = {dashLength, spacingForLineLength(dashRect.size.width)};
```

Comment:
The code in this block and the code in the following block look very similar.  Any chance these two can be parametrized into a single method?